File: android-v0-compose/app/src/main/java/com/example/v0clone/data/model/MessageModels.kt
package com.xlwl.AiMian.data.model

/**
 * 消息中心 - 会话摘要
 */
data class MessageSummary(
    val id: String,
    val title: String,
    val summary: String?,
    val type: String,
    val status: String,
    val unreadCount: Int,
    val lastActivityAt: String,
    val lastReadAt: String?,
    val createdAt: String,
    val updatedAt: String,
    val latestEntry: MessageEntry?
)

/**
 * 消息中心 - 消息详情
 */
data class MessageDetail(
    val id: String,
    val title: String,
    val summary: String?,
    val type: String,
    val status: String,
    val unreadCount: Int,
    val lastActivityAt: String,
    val lastReadAt: String?,
    val createdAt: String,
    val updatedAt: String,
    val entries: List<MessageEntry>
)

/**
 * 会话内单条消息
 */
data class MessageEntry(
    val id: String,
    val senderType: String,
    val senderId: String?,
    val senderName: String?,
    val content: String,
    val metadata: Map<String, Any>?,
    val createdAt: String
)

/**
 * 创建留言
 */
data class CreateMessageRequest(
    val title: String,
    val content: String,
    val type: String? = null
)

/**
 * 回复留言
 */
data class ReplyMessageRequest(
    val content: String,
    val metadata: Map<String, Any>? = null
)

/**
 * 标记已读响应
 */
data class MessageReadResponse(
    val id: String,
    val status: String,
    val unreadCount: Int,
    val lastReadAt: String?
)

File: iOS-v0/Sources/Models/MessageModels.swift
import Foundation

struct MessageSummary: Decodable, Identifiable, Sendable {
  let id: String
  let title: String
  let summary: String?
  let type: String
  let status: String
  let unreadCount: Int
  let lastActivityAt: String
  let lastReadAt: String?
  let createdAt: String
  let updatedAt: String
  let latestEntry: MessageEntry?
}

struct MessageDetail: Decodable, Identifiable, Sendable {
  let id: String
  let title: String
  let summary: String?
  let type: String
  let status: String
  let unreadCount: Int
  let lastActivityAt: String
  let lastReadAt: String?
  let createdAt: String
  let updatedAt: String
  let entries: [MessageEntry]
}

struct MessageEntry: Decodable, Identifiable, Hashable, Sendable {
  let id: String
  let senderType: String
  let senderId: String?
  let senderName: String?
  let content: String
  let metadata: [String: JSONValue]?
  let createdAt: String
}

struct CreateMessageRequest: Encodable, Sendable {
  let title: String
  let content: String
  let type: String?
}

struct ReplyMessageRequest: Encodable, Sendable {
  let content: String
  let metadata: [String: JSONValue]?
}

struct MessageReadResponse: Decodable, Sendable {
  let id: String
  let status: String
  let unreadCount: Int
  let lastReadAt: String?
}
